source/digits_hits/include/GateGlobalActorHit.hh

```diff
@@ -0,0 +1,147 @@
+  GateGlobalActorHit( const G4Step& step, const std::string& volume_name );
+  GateGlobalActorHit( const GateGlobalActorHit& hit );
+  ~GateGlobalActorHit();
+
+  //get and set methods
+
+  void setVolumeName( const std::string& volume_name );
+  std::string getVolumeName() const;
+
+  void setScintillatorPosition( const TVector3& position );
+  TVector3 getScintillatorPosition() const;
+
+  void setEventID( const G4int& event_id );
+  G4int getEventID() const;
+
+  void setTrackID( const G4int& track_id );
+  G4int getTrackID() const;
+
+  void setEnergyBeforeProcess( const G4double& energy );
+  G4double getEnergyBeforeProcess() const;
+
+  void setEnergyAfterProcess( const G4double& energy );
+  G4double getEnergyAfterProcess() const;
+
+  void setEnergyLossDuringProcess( const G4double& energy );
+  G4double getEnergyLossDuringProcess() const;
+
+  void setMomentumDirectionBeforeProcess( const TVector3& momentum );
+  TVector3 getMomentumDirectionBeforeProcess() const;
+
+  void setMomentumDirectionAfterProcess( const TVector3& momentum );
+  TVector3 getMomentumDirectionAfterProcess() const;
+
+  void setProcessPosition( const TVector3& position);
+  TVector3 getProcessPosition() const;
+
+  void setEmissionPointFromSource( const TVector3& position);
+  TVector3 getEmissionPointFromSource() const;
+
+  void setEmissionMomentumDirectionFromSource( const TVector3& momentum );
+  TVector3 getEmissionMomentumDirectionFromSource() const;
+
+  void setEmissionEnergyFromSource( const G4double& energy );
+  G4double getEmissionEnergyFromSource() const;
+
+  void setParticleName( const std::string& particle_name );
+  std::string getParticleName() const;
+
+  void setParticlePGDCoding( const G4int& pgd_code );
+  G4int getParticlePGDCoding() const;
+
+  void setPolarizationBeforeProcess( const TVector3& polarization );
+  TVector3 getPolarizationBeforeProcess() const;
+
+  void setPolarizationAfterProcess( const TVector3& polarization );
+  TVector3 getPolarizationAfterProcess() const;
+
+  void setProcessName( const std::string& process_name );
+  std::string getProcessName() const;
+
+  void setParentID( const G4int& parent_id );
+  G4int getParentID() const;
+
+  void setInteractionTime( const G4double& time );
+  G4double getInteractionTime() const;
+
+  void setLocalTime( const G4double& time );
+  G4double getLocalTime() const;
+
+  void setGlobalTime( const G4double& time );
+  G4double getGlobalTime() const;
+
+  void setProperTime( const G4double& time );
+  G4double getProperTime();  
+
+  // comparing methods
+  bool isTheSameScintillator( const GateGlobalActorHit& hit ) const;
+  bool isTheSameEventID( const GateGlobalActorHit& hit ) const;
+  bool isTheSameTrackID( const GateGlobalActorHit& hit ) const;
+
+ private:
+  void extractDataFromStep( const G4Step& step );
+  TVector3 getTVector3( const G4ThreeVector& input_vector ) const;
+  G4double keV( const G4double& energy ) const;
+
+ private:
+  std::string mVolumeName;
```

Comment:
I would initialize all the fields you can here.

Reply:
Done